include/aikido/util/yaml_utils.hpp

```diff
@@ -0,0 +1,202 @@
+inline void deserialize(YAML::Node const &node,
+                        Eigen::Transform<_Scalar, Dim, Mode, _Options> &pose)
+{
+    deserialize(node, pose.matrix());
+}
+
+template <class T>
+inline bool has_child(YAML::Node const &node, T const &key)
+{
+	return node[key].IsDefined();
+}
+
+} // namespace util
+} // namespace aikido
+
+
+namespace YAML {
+namespace detail {
+
+template <typename MatrixType, bool IsVectorAtCompileTime>
+struct encode_impl {
+  static Node encode(MatrixType const &matrix)
+  {
+    assert(false && "Unknown MatrixType.");
+  }
+};
+
+template <typename MatrixType>
+struct encode_impl<MatrixType, true> {
+  static Node encode(MatrixType const &matrix)
+  {
+    typedef typename MatrixType::Index Index;
+
+    Node node;
+    node.SetTag("Vector");
+
+    for (Index i = 0; i < matrix.size(); ++i) {
+      node.push_back(Node(matrix[i]));
+    }
+    return node;
+  }
+};
+
+template <typename MatrixType>
+struct encode_impl<MatrixType, false> {
+  static Node encode(MatrixType const &matrix)
+  {
+    typedef typename MatrixType::Index Index;
+
+    Node node;
+    node.SetTag("Matrix");
+
+    for (Index r = 0; r < matrix.rows(); ++r)
+    {
+      Node row(NodeType::Sequence);
+
+      for (Index c = 0; c < matrix.cols(); ++c)
+      {
+          row.push_back(matrix(r, c));
+      }
+
+      node.push_back(row);
+    }
+    return node;
+  }
+};
+
+} //detail
+
+
+template <typename _Scalar, int _Dim, int _Mode, int _Options>
+struct convert<Eigen::Matrix<_Scalar, _Dim, _Mode, _Options> > {
+    typedef Eigen::Matrix<_Scalar, _Dim, _Mode, _Options> MatrixType;
+
+    static Node encode(MatrixType const &matrix)
+    {
+        YAML::Node node(NodeType::Sequence);
+        return detail::encode_impl<
+          MatrixType, MatrixType::IsVectorAtCompileTime>::encode(matrix);
+    }
+
+    static bool decode(
+        YAML::Node const &node,
+        Eigen::Matrix<_Scalar, _Dim, _Mode, _Options> &matrix)
+    {
+        aikido::util::deserialize(node, matrix);
+
+        if (node.Tag() == "Vector" || node.Tag() == "!Vector"
+         || node.Tag() == "Matrix" || node.Tag() == "!Matrix") {
+            aikido::util::deserialize(node, matrix);
+            return true;
+        } else {
+            return false;
+        }
+    }
+};
+
+template <typename _Scalar, int _Dim, int _Mode, int _Options>
+struct convert<Eigen::Transform<_Scalar, _Dim, _Mode, _Options> > {
+    typedef Eigen::Transform<_Scalar, _Dim, _Mode, _Options> TransformType;
+    typedef typename TransformType::MatrixType MatrixType;
+
+    static Node encode(TransformType const &transform)
+    {
+        return convert<MatrixType>::encode(transform.matrix());
+    }
+
+    static bool decode(Node const &node, TransformType &transform)
+    {
+        return convert<MatrixType>::decode(node, transform.matrix());
+    }
+};
+
+template <class T>
+inline void operator>>(Node const &node, T &value)
+{
+    value = node.as<T>();
+}
```

Comment:
I'd like to remove this. If I remember correctly, I added this as a workaround to support both `yaml-cpp` versions 0.3 (which used the bitshift operators) and 0.5 (which used the `as` function).